specs/_features/ePBS/builder.md

```diff
@@ -0,0 +1,229 @@
+
+#### Constructing the `SignedExecutionPayloadEnvelope`
+
+1. Build `ExecutionPayloadEnvelope`
+- Set `payload`  to the saved `execution_payload` from constructing the header.
+- Set `builder_index` to the index of the builder.
+- Set `beacon_block_root` to the head block root.
+- Set `blob_kzg_commitments` to the kzg commitments from the saved blobs bundle `blobs_bundle.kzg_commitments`.
+- Set `state_root` to the state root after processing head block.
+
+2. Sign the `ExecutionPayloadEnvelope` with the builder's private key
+
+```python
+def get_execution_payload_envelope(state: BeaconState, payload: ExecutionPayloadEnvelope, privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_BEACON_BUILDER, compute_epoch_at_slot(state.slot))
+    signing_root = compute_signing_root(payload, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+3. Build `SignedExecutionPayloadEnvelope`
+- Set `message` to the `ExecutionPayloadEnvelope` from step 1
+- Set `signature` to the `BLSSignature` from step 2
+
+#### Broadcast execution payload envelope
+
+Finally, the validator broadcasts `signed_execution_payload_envelope` to the global `execution_payload_envelope` pubsub topic.
+
+#### Broadcast blob sidecars
+
+`SignedBlobSidecars`s constructions remain the same as in the deneb spec. The builder will broadcast `SignedBlobSidecars` to the global `blob_sidecars` pubsub topic the same time as `SignedExecutionPayloadEnvelope` is broadcasted.
+Builder may broadcast `SignedBlobSidecars` before `SignedExecutionPayloadEnvelope` if it is safe to reveal. (ie. the head block contains the payload header)
+
+## Design Decision Rationale
+
+### What are the changes with regard to blob sidecars?
+- Builder does not have to commit KZG commitments at the header phase. Given transaction contains versioned hashes, it's committed implicitly there given the transaction root.
+- This means KZG will only be known at the payload phase, and the commitments are moved from block body to the execution payload envelope.
+- Builder will gossip blob sidecars at the same time as the execution payload envelope. This is to ensure that the blob sidecars are available to the proposer when the proposer is building the block.
+- Builder may gossip the blob sidecars earlier than the execution payload.
+
+### How does builder to proposer payment work?
+
+- Builder to proposer payment is unconditional after processing the signed execution payload header and the block remains head and canonical throughout. The builder will not lose the bid if the block is reorged.
+
+### Should the builder worry about same slot unbundling?
+
+- There are cases to consider for same slot unbundling. the common case is proposer equivocate and broadcast the equivocated block to the network after builder has revealed the payload. In this case, the next slot proposer will also have to collude and build on the equivocated block.
```

Comment:
Not only the next slot proposer but also a vast majority of the committee since attesters would have voted for the previous head.